criblio/appscope
# Update the `scope start` command (#1414)

**This PR**
Updates the `scope start` command so that it only:
- installs libscope.so
- is able to install in a remote namespace (host, child or parent container)
- renames `--prefix` to `--rootdir` in all applicable cli commands

**Testing**
The scope start integration test has been commented out until filter functionality has been fully migrated.
There is a cli start integration test that runs the command and checks for an expected return code.

**Documentation**
The documentation for the `scope start` command has been updated in the cli reference document.

**QA Instructions**
```
scope start
sudo scope start
sudo scope start --rootdir /path/to/host # from a container
```
The library should be installed.
In the case where --rootdir is specified, the library should be installed in that remote namespace.
In the case where root user/sudo was user, the library should be installed in /usr/lib/appscope, otherwise /tmp/appscope
Try to install libscope.so locally, in a child namespace, and in a parent namespace, using this command.
Use the sudo ./cleanup_after_start.sh script in between runs.
Container-to-container example:
```
docker run --rm -it -v /home/sean/src/appscope/:/opt/appscope -v /:/hostfs:ro --privileged ubuntu # origin container
docker run --rm -it ubuntu # target container
# in origin container
./bin/linux/x86_64/scope start --rootdir /hostfs/proc/1559501/root # where 1559501 is the pid from the host perspective
```
